### ISSUE TITLE
`govendor update thepackage/...` and all it's dependencies? 

### ISSUE THREAD

**Issue by slimsag** [kardianos/govendor#138]
I've searched and searched through the README.md and the FAQ, but was surprised that there is not an answer for this use case given that it seems like it would be extremely common.

The use case: I have a package `thepackage` which has a huge amount of dependencies (more than I can / want to track by hand). I vendor that package, and now I want to update that package and it's all of it's dependencies.

I've found `govendor update thepackage/...` updates `thepackage` and all packages that are under `thepackage`

But how can you ask govendor to also update any dependencies that `thepackage` references?

**Comment by kardianos** [comment]
Use fetch

**Comment by kardianos** [comment]
Actually, I'm not sure but I'll check.

**Comment by slimsag** [comment]
Thanks @kardianos -- I tried just now with the `+external` flag but to no avail:

```
$ govendor fetch -v +external sourcegraph.com/sourcegraph/appdash/...
Get latest revision "sourcegraph.com/sourcegraph/appdash"
Get latest revision "sourcegraph.com/sourcegraph/appdash/httptrace"
Get latest revision "sourcegraph.com/sourcegraph/appdash/internal/wire"
Get latest revision "sourcegraph.com/sourcegraph/appdash/sqltrace"
Get latest revision "sourcegraph.com/sourcegraph/appdash/traceapp"
Get latest revision "sourcegraph.com/sourcegraph/appdash/traceapp/tmpl"
```

Likewise `git status` shows it only updated packages under the given import path filter, not any others.

It would be nice if it could be done with `update` instead of `fetch` too, because hitting the network for updates is not always desirable (I know the versions in my $GOPATH are right, but not certain the latest remote versions are right).

**Comment by kardianos** [comment]
@slimsag oh, I know why fetch did not get latest for all deps in this case, they were already in GOPATH or vendor folder and they were not explicitly specified.

It looks like I need an argument to declare update/add/fetch for this package AND all it's (non-stdlib) dependencies. This feature has been requested several times by various people recently.

**Comment by slimsag** [comment]
Yep that sounds like it would do the trick exactly!

**Comment by kardianos** [comment]
Proposal:
In the update, add, remove, and fetch sub-commands, add the flag "-R" to recursively perform the action on all that packages deps to. For remove, it would only remove the packages that are not referenced elsewhere.

**Comment by kaedys** [comment]
> For remove, it would only remove the packages that are not referenced elsewhere.

Further, make sure this works with `govendor remove -R +unused`, to recursively remove all unused packages, and then any packages that are now unused with their removal, and so on, until the only packages that are remaining are those explicitly used by the root repo, or packages those reference.  Basically works as a "clean up my `/vendor`" command.

**Comment by kardianos** [comment]
@kaedys Good point. I can add a test for that too.

**Comment by timoreimann** [comment]
@kardianos any update on this one? It's likely the one key feature missing to me. Would love to have it!

**Comment by kardianos** [comment]
Not yet. This last month I've had other activities on my plate. It looks like most of them are taken care of so I can focus on this soon.

**Comment by timoreimann** [comment]
Sounds promising. Appreciate the update?

**Comment by timoreimann** [comment]
It's been a while. :-) Is there still a chance for this feature getting implemented?

I can imagine the prospected, long-term shift towards golang/dep may have changed plans somehow?

**Comment by kardianos** [comment]
No, just lack of time. Been busy at work / home. Sorry. I try to keep up with PRs, but haven't spent too much time on features.

**Comment by glasser** [comment]
Does anyone have a workaround for this?

**Comment by slimsag** [comment]
The workaround I have is to find all the dependencies of your package (manually, or via starting with a clean GOPATH and `go get`'ing it, or via `go list` + filtering) and then running `govendor update thepackage/...` for each one AFAIK.

**Comment by BugDiver** [comment]
`govendor update ./...` adds local packages as dependencies which create problems. Any idea how to avoid this?

**Comment by wangming1993** [comment]
+1
